### PR TITLE
Add a script that can add all clusters as datasources for central grafana

### PR DESCRIPTION
**Pull request by GeorgianaElena**
This adds a script that updates the datasources of our central grafana (the 2i2c one) to encompass all authenticated prometheus instances of our clusters. It also adds more colors to the `print_color` function.

It can be ran every time we add a new cluster, or deploy the support chart for an existing one.

Note: 
1. I've ran this successfully
2. The clusters that failed to be added for reasons that were expected were:

datasource-updater-output

Ref: https://github.com/2i2c-org/infrastructure/issues/328

### CODE REVIEW

File: deployer/update_grafana_datasources.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+import requests
+from file_acquisition import find_absolute_path_to_cluster_file, get_decrypted_file
+from helm_upgrade_decision import get_all_cluster_yaml_files
+from ruamel.yaml import YAML
+from utils import print_colour
+
+yaml = YAML(typ="safe")
+
+DATASOURCE_ENDPOINT = "https://grafana.pilot.2i2c.cloud/api/datasources"
```
Comment (yuvipanda) [inline deployer/update_grafana_datasources.py:22]: Let's pass these in as commandline parameters with these set as defaults.

File: deployer/update_grafana_datasources.py [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+
+    print(f"Successfully retrieved the datasources!")
+    datasources = response.json()
+    return [datasource["name"] for datasource in datasources]
+
+
+def main():
+    # Get a list of the clusters that already have their prometheus instances used as datasources
+    datasources = get_clusters_used_as_datasources()
+    print(datasources)
```
Comment (yuvipanda) [inline deployer/update_grafana_datasources.py:168]: leftover debugging line?

**Review by yuvipanda — CHANGES_REQUESTED** [review]
Minor style nits, but otherwise yay!

We should run this every time we create a new cluster, so let's add that to the docs? When else should this be run?

Thank you for working on this!

**Comment by GeorgianaElena** [comment]
Thanks for the feedback @yuvipanda 🚀 ! I believe I have addressed all the comments in the last few commits. Let me know what you think 👀 !

File: deployer/update_grafana_datasources.py [hunk 3]
```diff
@@ -143,31 +139,54 @@ def build_request_headers():
...
 
 
 def main():
+    argparser = argparse.ArgumentParser(
+        description="""A command line tool to update Grafana
+        datasources.
+        """
+    )
+
+    argparser.add_argument(
```
Comment (yuvipanda) [inline deployer/update_grafana_datasources.py:165]: set the defaults here to be the 2i2c cluster's grafana?

File: deployer/update_grafana_datasources.py [hunk 4]
```diff
@@ -143,31 +139,54 @@ def build_request_headers():
...
+        "cluster_name", type=str, help="The name of the cluster where the Grafana lives"
+    )
+
+    argparser.add_argument("grafana_url", type=str, help="The public URL of Grafana")
```
Comment (yuvipanda) [inline deployer/update_grafana_datasources.py]: Can we read this in from the cluster config too? I think we have it under ingress.

**Review by yuvipanda — APPROVED** [review]
Two minor comments, feel free to merge after that.

Thanks a lot @GeorgianaElena!

File: deployer/update_grafana_datasources.py [hunk 5]
```diff
@@ -0,0 +1,240 @@
+"""
+### Summary
+
+This is a helper script that will make sure the central grafana at https://grafana.pilot.2i2c.cloud is configured to use as datasource the authenticated prometheus instances of all the clusters that we run.
+
+### Run the script
+
+$ python deployer/grafana_datasources_manager.py
```
Comment (choldgraf) [inline deployer/update_grafana_datasources.py:8]:
```suggestion
### Summary

Ensures that the central grafana at https://grafana.pilot.2i2c.cloud is configured to use as datasource the authenticated prometheus instances of all the clusters that we run.

### How to use

This is meant to by run as a script from the command line, like so:

$ python deployer/grafana_datasources_manager.py
```

File: deployer/update_grafana_datasources.py [hunk 6]
```diff
@@ -0,0 +1,240 @@
+"""
+### Summary
```
Comment (choldgraf) [inline deployer/update_grafana_datasources.py:2]: Could rename this to something like `update_central_grafana_datasources.py`, or put this in a folder that makes it clear it only applies to the central grafana and not each support chart grafana?

**Review by choldgraf — APPROVED** [review]
Just a few quick documentation notes on this, thanks so much @GeorgianaElena this looks really nice :-) I am excited to start using this central Grafana!!!

File: deployer/update_grafana_datasources.py (hunk 3 above)
Comment (GeorgianaElena) [reply deployer/update_grafana_datasources.py:165]: I did forget to set defaults, but did it in a later commit https://github.com/2i2c-org/infrastructure/pull/1215/commits/2e1e3c3d05ddf1abafaccfd28bf994c6f9a43231

File: deployer/update_grafana_datasources.py (hunk 6 above)
Comment (GeorgianaElena) [reply deployer/update_grafana_datasources.py]: Done in https://github.com/2i2c-org/infrastructure/pull/1215/commits/6f0c993e444bb7f6b6dbd9890bb808ea2f341d27

File: deployer/update_grafana_datasources.py (hunk 4 above)
Comment (GeorgianaElena) [reply deployer/update_grafana_datasources.py]: Done in https://github.com/2i2c-org/infrastructure/pull/1215/commits/798e53f070b2662acb62404e11a39e140f67d025

File: docs/howto/operate/grafana.md [hunk 7]
```diff
@@ -2,6 +2,11 @@
 # Grafana Dashboards
 
 Each 2i2c Hub is set up with [a Prometheus server](https://prometheus.io/) to generate metrics and information about activity on the hub, and each cluster of hubs has a [Grafana deployment](https://grafana.com/) to ingest and visualize this data.
+
+```{note}
+The Grafana deployement in the `2i2c` cluster ingests data from all the 2i2c clusters and will soon be able to be used as "the central Grafana".
```
[the remark this answers is not in the document]
Comment (GeorgianaElena) [reply docs/howto/operate/grafana.md]: Done in https://github.com/2i2c-org/infrastructure/pull/1215/commits/b13d590ba0c3e7ef714b8ab5a27d231a95dda114

**Comment by GeorgianaElena** [comment]
Thanks everyone for the feedback ✨ I believe I have addressed all the comments and this is now ready to merge 🚀

**Pull request merged**